### PR TITLE
Feature- template for notifications page

### PR DESCRIPTION
**Pull request by gbudjeakp**

### CODE REVIEW

File: client/src/components/FlightSearch.js [hunk 1]
```diff
@@ -73,68 +71,70 @@ const FlightSearch = ({ submit }) => {
...
-  				flights: data,
-  			});
+    // if (from && to && departureDate && returnDate) {
+    // 	if (returnDate >= departureDate) {
+    // 		const response = await fetch(
+    // 			`api/flights/quotes/${fromCity}/${toCity}/${formattedDepartureDate}/?inboundDate=${formattedReturnDate}`
+    // 		);
+    // 		const data = await response.json();
+    // 		if (data.status === 200) {
+    // 			submit({
```
Comment (johnsciutto) [inline client/src/components/FlightSearch.js]: Delete the code you're commenting out. If you need to get it back, you can always do that with `git` if it was part of a previous commit.

File: client/src/components/FlightSearch.js [hunk 2]
```diff
@@ -32,35 +32,33 @@ const FlightSearch = ({ submit }) => {
...
-
   const handleFromLocation = async (event, value) => {
     if (value === "") {
       return;
     }
     try {
       const response = await fetch(`/api/flights/places/${value}`);
       const data = await response.json();
-	  console.log(data)
+      console.log(data);
```
Comment (johnsciutto) [inline client/src/components/FlightSearch.js]: Delete the `console.log`s that are just for checking things out.

File: client/src/components/FlightSearch.js [hunk 3]
```diff
@@ -73,68 +71,70 @@ const FlightSearch = ({ submit }) => {
...
+    event.preventDefault();
+    const fromCity = fromCities.find((city) => from === city.PlaceName).PlaceId;
+    const toCity = toCities.find((city) => to === city.PlaceName).PlaceId;
+    const formattedDepartureDate = formatDate(departureDate);
+    const formattedReturnDate = formatDate(returnDate);
+    const response = await fetch(
+      `api/flights/quotes/${fromCity}/${toCity}/${formattedDepartureDate}/?inboundDate=${formattedReturnDate}`
+    );
+    const data = await response.json();
+	console.log(data)
```
Comment (johnsciutto) [inline client/src/components/FlightSearch.js]: Same as above

File: client/src/components/FlightSearch.js (hunk 1 above)
Comment (gbudjeakp) [reply client/src/components/FlightSearch.js]: Will effect those changes now. Forgot to pull from dev before pushing.

File: client/src/components/FlightSearch.js [hunk 4]
```diff
@@ -73,81 +82,69 @@ const FlightSearch = ({ submit }) => {
...
+			const data = await response.json();
+			if (response.status === 200) {
+				submit({ flights: data });
+			} else if ("from" in data) {
+				setFromError(true);
+			} else if ("to" in data) {
+				setToError(true);
+			} else if ("outboundDate" in data) {
+				setDepartDateError(true);
+			}
```
Comment (johnsciutto) [inline client/src/components/FlightSearch.js]: There's quite a bit of duplicated code here. Check to see if you can reduce it 🙂

Comment (newCodeWriter) [reply client/src/components/FlightSearch.js]: This is my code. I found the way to reduce it was to isolate the fetch call into its own function. Please see below: 

	const handleFetch = async () => {
		const fromCity = cities.find((city) => from === city.title).title;
		const toCity = cities.find((city) => to === city.title).title;
		const formattedDepartureDate = formatDate(departureDate);
		const formattedReturnDate = returnDate ? formatDate(returnDate) : null;
		const response = await fetch(
			`api/flights/quotes/${fromCity}/${toCity}/${formattedDepartureDate}${
				returnDate ? `/?inboundDate=${formattedReturnDate}` : ""
			}`
		);
		const data = await response.json();
		if (response.status === 200) {
			submit({ flights: data });
		} else if ("from" in data) {
			setFromError(true);
		} else if ("to" in data) {
			setToError(true);
		} else if ("outboundDate" in data) {
			setDepartDateError(true);
		} else if ("inboundDate" in data) {
			setReturnDateError(true);
		}
	};
	const handleSubmit = (event) => {
		event.preventDefault();
		if (from && to && from !== to && departureDate && returnDate) {
			if (returnDate >= departureDate) {
				handleFetch();
			} else {
				setReturnDateError(true);
			}
			// return date is optional
		} else if (from && to && from !== to && departureDate) {
			handleFetch();
		} else if (from === to || !to) {
			setToError(true);
		} else if (!from) {
			setFromError(true);
		}
	};

Comment (gbudjeakp) [reply client/src/components/FlightSearch.js]: Thanks @newCodeWriter.

File: client/src/pages/Notifications.js [hunk 5]
```diff
@@ -1,13 +1,38 @@
...
-    return (
-        <div>
-            <h1 style={{marginLeft: 390}}>Notifications are here</h1>
-        </div>
-    )
+  //Flight variables
+  const departureDate = new Date("March 30, 2021");
+  const currentDate = new Date();
+  const hotel = "Scrumptious"
+  const  Destination ="Toronto"
```
Comment (johnsciutto) [inline client/src/pages/Notifications.js]: I know this is just mock data, but remember to lowercase the destination if you're just making it a variable and not a class/constructor 🙂

**Review by johnsciutto — COMMENTED** [review]
Looks good to me 🙂

**Review by BoomBoomRay — APPROVED** [review]
Once all the comments have been fixed. Approved.

File: client/src/pages/Notifications.js (hunk 5 above)
Comment (gbudjeakp) [reply client/src/pages/Notifications.js]: Oops... thanks for that. I'll effect the change immediately. I didn't see it

**Pull request closed without merging**